File: blormlib/src/main/java/br/com/bloder/blormlib/validation/validations/Checked.java
package br.com.bloder.blormlib.validation.validations;

import android.widget.CheckBox;

import br.com.bloder.blormlib.validation.Validate;
import br.com.bloder.blormlib.validation.Validation;

public class Checked extends Validation {

    private CheckBox checkBox;

    @Override
    public Validate validate() {
        return new Validate() {
            @Override
            public boolean validate() {
               checkBox = (CheckBox) field;
               return checkBox.isChecked();
            }

            @Override
            public void onError() {
                checkBox.setError("This field can't be unchecked");
            }

            @Override public void onSuccess() {
                checkBox.setError(null);
            }
        };
    }
}

File: blormlib/src/main/java/br/com/bloder/blormlib/validation/validations/Checked.kt
package br.com.bloder.blormlib.validation.validations

import android.widget.CheckBox
import br.com.bloder.blormlib.validation.Validate
import br.com.bloder.blormlib.validation.Validation

/**
 * Created by bloder on 03/09/16.
 */

class Checked : Validation(), Validate {

    private var checkBox: CheckBox? = null
    override fun validateIt(): Boolean {
        checkBox = field as CheckBox
        return checkBox?.isChecked as Boolean
    }
    override fun onError() { checkBox?.error = "This field can't be unchecked" }
    override fun onSuccess() { checkBox?.error = null }
    override fun validate(): Validate { return this }
}